src/python/pants/goal/context.py

```diff
@@ -399,3 +400,14 @@ def execute_process_synchronously(self, execute_process_request, name, labels=No
       workunit.output("stderr").write(result.stderr)
       workunit.set_outcome(WorkUnit.FAILURE if result.exit_code else WorkUnit.SUCCESS)
       return result
+
+  def execute_process_synchronously_or_raise(self, execute_process_request, name, labels=None):
```

Comment:
Maybe rename the other to `fallibly_execute_process_synchronously` or `execute_process_synchronously_without_raising` or similar, so it's obvious that if you don't want that behaviour you should look for another function?